bacon/model/StorageCouchBaseDB.swift

```diff
@@ -160,6 +165,40 @@ class StorageCouchBaseDB {
         }
     }
 
+    func deleteTransaction(_ transaction: Transaction) throws {
+        // Fetch the specific document from database
+        guard let transactionId = transactionMapping[transaction] else {
+            throw StorageError(message: """
+                Unable to find mapping of transaction object to it's unique id in the database.
```

Comment:
Typo: should be its* not it's haha

Also, should this ever happen? O.o If not, should it be a fatal error instead?

Also I think this part should be logged.

Reply:
This will happen if the transaction requesting to be deleting was not retrieved through loading of database.
Meaning if someone creates a transaction and requests it to be deleted. The database would not have the mapping of the id and the transaction object.
Hence, I was asking if every handling of transaction go through the load method.

Reply:
I don't think encountering problem deleting a transaction warrants a need to crash the application?

Reply:
I'm not sure... I feel like if that (i.e. directly creating a transaction and requesting for it to be deleted) is something that should never happen, then if that happens it should be a fatal error right?

Because it's not like file not found error, or network issues, that can be gracefully handled. This kinda thing is more like programming error... not too sure @kumuwu what do you think?